tests/src/Tables/Concerns/CanSearchRecordsTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+use Filament\Tests\Tables\TestCase;
+
+use function PHPUnit\Framework\assertCount;
+
+uses(TestCase::class);
+
+/**
+ * The user-entered search string for a table is split into words
+ * on any runs of whitespace.
+ */
+
+it('can split on whitespace', function () {
+    $trait = new class {
+        use \Filament\Tables\Concerns\CanSearchRecords {
+            extractTableSearchWords as public;
+        }
+    };
+
+    assertCount(1, $trait->extractTableSearchWords('test'));
+
+    assertCount(2, $trait->extractTableSearchWords('testy test'));
+    assertCount(2, $trait->extractTableSearchWords('testy   test'));
+    assertCount(2, $trait->extractTableSearchWords("testy \t \n \r  test"));
+
+    // Leading and trailing whitespace are extracted as empty words.
+    assertCount(3, $trait->extractTableSearchWords('  test  '));
+    assertCount(3, $trait->extractTableSearchWords(" \t test \n "));
+    assertCount(4, $trait->extractTableSearchWords('  testy test  '));
+    assertCount(4, $trait->extractTableSearchWords(" \t testy test \n "));
+    assertCount(5, $trait->extractTableSearchWords("   \t testy  \t  tasty  \n  test \r   "));
+
+    $this->assertSame(['', 'test', ''], $trait->extractTableSearchWords('   test   '));
```

Comment:
Please explain this behaviour, why should words be created out of whitespace, when they're just `''`?

Reply:
That's the consequence of not trimming in the `extractTableSearchWords()` method being tested. The test just shows it explicitely. It's not desired behaviour IMO.

Reply:
An empty string would also return `['']` which is probably also undesirable.

Reply:
Yes, I don't believe that there should ever be empty strings generated like this, the regex should match **one or more** whitespace characters maybe?